components/common-go/baseserver/options.go

```diff
@@ -38,111 +35,118 @@ type config struct {
 	grpcHealthCheck grpc_health_v1.HealthServer
 }
 
-func defaultConfig() *config {
-	return &config{
-		logger:          log.New(),
-		hostname:        "localhost",
-		httpPort:        -1, // disabled by default
-		grpcPort:        -1, // disabled by default
-		debugPort:       9500,
+func defaultOptions() *options {
+	return &options{
+		logger: log.New(),
+		config: &Configuration{
+			Services: ServicesConfiguration{
+				GRPC: nil, // disabled by default
+				HTTP: nil, // disabled by default
+			},
+		},
+
 		closeTimeout:    5 * time.Second,
 		healthHandler:   healthcheck.NewHandler(),
 		metricsRegistry: prometheus.NewRegistry(),
 		grpcHealthCheck: &GrpcHealthService{},
 	}
 }
 
-type Option func(cfg *config) error
+type Option func(opts *options) error
 
-func WithHostname(hostname string) Option {
-	return func(cfg *config) error {
-		cfg.hostname = hostname
+// WithConfig uses a config struct to initialise the services
+func WithConfig(config *Configuration) Option {
+	return func(opts *options) error {
+		opts.config = config
 		return nil
 	}
 }
 
-// WithHTTPPort sets the port to use for an HTTP server. Setting WithHTTPPort also enables an HTTP server on the baseserver.
-func WithHTTPPort(port int) Option {
-	return func(cfg *config) error {
-		cfg.httpPort = port
+func WithoutBuiltinServices() Option {
```

Comment:
IMO, this feel like it shouldn't really be an option. I think that all of our servers should have a debug server running and until we've come across a use-case where we definitely do need it, I'd not even offer the option.

Reply:
Agreed - it's only used for testing. I'd be happy to remove the option and just make the tests set the flag directly.

Reply:
Agreed - will remove